SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -1932,7 +1934,7 @@ void CDriver::Numerics_Preprocessing(CConfig *config, CGeometry **geometry, CSol
         break;
       case SPACE_UPWIND :
         for (iMGlevel = 0; iMGlevel <= config->GetnMGLevels(); iMGlevel++) {
-          if (spalart_allmaras || neg_spalart_allmaras || e_spalart_allmaras || comp_spalart_allmaras || e_comp_spalart_allmaras ) {
+          if (spalart_allmaras || spalart_allmaras_noft2 || neg_spalart_allmaras || neg_spalart_allmaras_noft2 || e_spalart_allmaras || comp_spalart_allmaras || e_comp_spalart_allmaras ) {
```

Comment:
This way of handling the new options is a no go on my part, if we do it like this we will be chasing SA bugs for months, exactly like it happened when the solver options changed. Even with SST_SUST which was just one new option I fixed a bug last month or so.

I trust you to have changed all places in the code, but other people will not be aware (because very few people follow the pull requests) and unfortunately there are 10's of branches that do not track develop closely.

So, as a bare minimum this "is SA", "is SA-neg" logic needs to be encapsulated in a single method of CConfig, and replaced everywhere.
If you want to take it further we should create a small lightweight class that condenses all SA info, including coefficients.

Actually even less work, and probably more appropriate, create a boolean option called SA_USE_FT2= YES/NO, and avoid changing all the enum's and turbulence options.

Reply:
I totally agree with that. I would suggest to follow the comment from @clarkpede on issue #992 so we are able to include more than one variation.
Maybe on the config file we can include a tag for all the SA variations. Something like:

SA_VARIATIONS: NO_FT2, COMP, MOD_VORT, etc

which are basically Boolean to identify whether we want to use no ft2, comp, modified vorticity, etc.

Then, on turb_sources to have only one class, the original SA model. Therein, include all possible variations.

Reply:
Yes that sounds reasonable for the options and there is precedent with the QCR modification of the stress tensor.

However, having classes that look like Swiss army knives by means of if's and switch's is a bad software practice IMO.
There are "modern" and very efficient ways to specialize such small aspects, I'm using (and have described) some in #1022, we can discuss them in the next developers meeting if you want.

Reply:
I will take a look to that!